### PR TITLE
Qlog data moved

### PR DESCRIPTION
**Pull request by LPardue**
Opening mainly to get some eyes on.

Qlog draft 02 has a data_moved event that allows logging when stream data is moved into or out of the transport. Logging this information allows us in the qvis multiplexing tool to see how the quiche-client reads data from the transport layer.

Our definition of data_moved is bit old, so this change tweaks it to contain the length. In testing I found that `SendBuf.off()` always returned `0` while the qvis expects the offset to move along at the stream offset grows. That is why I added the `cumulative_offset()` function as a hack. Happy to hear a better solution to this.

Right now data_moved is in the category HTTP/3 but that will change to be transport soon. So we can rename things in a separate followup PR.

### CODE REVIEW

File: src/stream.rs [hunk 1]
```diff
@@ -769,6 +769,11 @@ impl RecvBuf {
         Ok(())
     }
 
+    /// Returns the cumulative offset of data buffered.
+    pub fn cumulative_off(&self) -> u64 {
```
Comment (ghedo) [inline src/stream.rs:773]: This name sounds wrong. It's not really "cumulative" as there's just one offset, so maybe just call it `off()`?

File: src/stream.rs [hunk 2]
```diff
@@ -1009,6 +1014,11 @@ impl SendBuf {
...
```
Comment (ghedo) [inline src/stream.rs:1018]: Same.

File: tools/qlog/src/lib.rs [hunk 3]
```diff
@@ -1456,6 +1456,7 @@ pub enum EventData {
     H3DataMoved {
         stream_id: String,
         offset: Option<String>,
+        length: Option<u64>,
```
Comment (ghedo) [inline tools/qlog/src/lib.rs:1459]: What's the logic behind using `String` in some places but `u64` in others? IIRC you said `String` is for big numbers that can't be encoded in JSON, but using both makes the API pretty confusing.

Not really related to this PR, but in the future we should just make all these fields `u64` and then tag the ones that should be encoded as string with the `#[serde(serialize_with = "u64_to_str")] ` attribute, where `u64_to_str` is a function we implement to automatically encode specific fields as string. This way at least the API exposed by the `qlog` crate doesn't leak limitations in the underlying encoding format.

Comment (LPardue) [reply tools/qlog/src/lib.rs:1459]: The qlog crate's datamodel follows closely the real qlog datamodel, which has inconsistencies like this throughout the schema. 

I believe these will be fixed across the board in qlog-02, so we can fix stuff up then and consider custom serialize/deserialize functions.

File: src/stream.rs (hunk 1 above)
Comment (LPardue) [reply src/stream.rs]: as per discussion, renamed the old `off()` method to `off_front()` and `cumulative_off()` to `off_back()`.

File: src/lib.rs [hunk 4]
```diff
@@ -2448,6 +2451,18 @@ impl Connection {
             self.streams.collect(stream_id, local);
         }
 
+        qlog_with!(self.qlog_streamer, q, {
```
Comment (ghedo) [inline src/lib.rs:2454]: Actually, wait, I just realized that this is in the transport code. I don't think we should be logging HTTP/3 events from here...

Comment (ghedo) [reply src/lib.rs:2454]: Ah nvm, I read the PR description now.

**Pull request merged**